Background Tasks: Async background tasks fail silently if soap client return false

fneumann opened the pull request:
See https://mantis.ilias.de/view.php?id=42183

This PR ads a fallback to SyncTaskManager if the soap client returns false.

fneumann committed:
Fix #42183: Async background tasks fail silently

mjansenDatabay commented on src/BackgroundTasks/Implementation/TaskManager/AsyncTaskManager.php:
```diff
@@ -55,9 +55,14 @@ public function run(Bucket $bucket): void
                 $session_id . '::' . $client_id,
             ));
+            if ($result === false) {
+                $DIC->logger()->root()->info("[BT] Calling Webserver returned false, fallback to sync version");
```
Shouldn't we use `warning` or `error` in this case?

fneumann replied:
This would create many entries, if soap is not active. @chfsx may decide this.

chfsx replied:
I suggest to use the component logger for BackgroundTask instead of root and then use warning.

fneumann committed:
Merge branch 'release_9' into fix9-mantis42183

fneumann committed:
Improve logging of async background task

fneumann committed:
Check enabled soap to prevent orphaned background task entries

mjansenDatabay commented:
One thing that I discovered in projects with time consumer async. background tasks.

1. It could be worth setting a response timeout of `0` by calling `$soap_client->setResponseTimeout(0);` (in `\ILIAS\BackgroundTasks\Implementation\TaskManager\AsyncTaskManager::run`).
2. A `ignore_user_abort(true);` could be set in `\ilSoapBackgroundTasksAdministration::runAsync`

As far as I can see, the SOAP call is "Fire & Forget" anyway. 

This approach will ensure some of the `false` or `Exception` cases will not occur anymore and the client/caller (ILIAS itself) does not unnecessarily wait.

fneumann committed:
Don't check the return value of the SOAP call

This will be false in case of time-consuming background tasks.
We have to keep the "fire and forget".

fneumann commented:
I reverted the fallback to sync execution if the soap call returns false. This is the case for time-consuming tasks and and would result in a double execution. The initial reason for this PR (failed call for root user due to missing acceptance of legal documents) is solved by https://github.com/ILIAS-eLearning/ILIAS/commit/38d246e153c2f13be49ac9c8ddc46b934d25e0b0.

Nevertheless, I think this PR is still useful regarding the change of the logger and the pre-check for activated SOAP.

chfsx commented:
Thanks a lot, I squashed to one commit and picked to other branches as well
